File: code/csharp/DesignPatterns/adapter/UniversalAdapter.cs
namespace Adapter;
class UniversalAdapter : IElectricalOutlet
{
    private MobilePhone _device;

    public UniversalAdapter(MobilePhone device)
    {
        _device = device;
    }

    public void Request()
    {
        Console.WriteLine("Converting electrical power to device");
        _device.SpecificRequest();
    }
}

File: code/typescript/adapter/UniversalAdapter.ts
import { ElectricalOutlet } from "./ElectricalOutlet";
import { MobilePhone } from "./MobilePhone";

export class UniversalAdapter implements ElectricalOutlet {
    private device: MobilePhone;

    constructor(device: MobilePhone) {
        this.device = device;
    }

    request(): void {
        console.log("Converting electrical power to device");
        this.device.specificRequest();
    }
}
